### PR TITLE
Added <?? optional retrieve for JSON serialization of optional parameters

### PR DESCRIPTION
**Pull request by mpurland**
Note: The definition of operator <?? is in Operadics. The operator definition was temporarily added to JSON.swift to satisfy the requirement until the PR in Operadics is merged.

See: https://github.com/typelift/Operadics/pull/4

### CODE REVIEW

File: Swiftz/JSON.swift [hunk 1]
```diff
@@ -8,6 +8,12 @@
 
 import Foundation
 
+/// Temporarily added <?? parameter until <?? is added to Operadics.
+infix operator <?? {
```
Comment (CodaFi) [inline Swiftz/JSON.swift]: Tag Operadics/master in the cartfile and run Carthage update.  I'll put out a proper release later tonight.

Comment (mpurland) [inline Swiftz/JSON.swift]: :+1:

**Comment by mpurland** [comment]
Should be good to go after tests are complete.

File: SwiftzTests/JSONSpec.swift [hunk 2]
```diff
@@ -294,3 +294,60 @@ extension JSONSpec {
...
+	let required2: String
+	let optional1: Bool?
+	let optional2: Bool?
+	
+	static func fromJSON(x : JSONValue) -> TestObject? {
+		let p1: String? = x <? "required1"
+		let p2: String? = x <? "required2"
+		let p3: Bool?? = x <?? "optional1"
+		let p4: Bool?? = x <?? "optional2"
+
```
Comment (CodaFi) [inline SwiftzTests/JSONSpec.swift]: Same thing about spacing between colon and identifier above ^^

Comment (mpurland) [inline SwiftzTests/JSONSpec.swift]: :white_check_mark:

**Comment by CodaFi** [comment]
And we're off

:boat:

**Comment by mpurland** [comment]
:+1:

**Pull request merged**